Core/include/Acts/Detector/DetectorVolume.hpp

```diff
@@ -296,13 +296,68 @@ class DetectorVolume : public std::enable_shared_from_this<DetectorVolume> {
       visitor(s);
     }
     for (const auto& p : portals()) {
-      p->visitSurfaces(std::forward<visitor_t>(visitor));
+      p->visitSurface(std::forward<visitor_t>(visitor));
     }
     for (const auto& v : volumes()) {
       v->visitSurfaces(std::forward<visitor_t>(visitor));
     }
   }
 
+  /// @brief Visit all reachable surfaces of the detector - non-const
+  ///
+  /// @tparam visitor_t Type of the callable visitor
+  ///
+  /// @param visitor will be called for each found surface,
+  /// it will be handed down to contained volumes and portals
+  template <ACTS_CONCEPT(SurfacePtrVisitor) visitor_t>
+  void visitSurfacePtrs(visitor_t&& visitor) {
+    for (auto& s : surfacePtrs()) {
+      visitor(s);
```

Comment:
Is there are strong need to have a separate visitor for the `shared_ptr`s? Could this line not just dereference or unpack the raw pointers?

Reply:
Yes, one is for const visit, one is for non-const visit, i.e. we can throw a way quite some code for material loading, this could just be a visitor.

Reply:
Changed.